zrml/orderbook/src/lib.rs

```diff
@@ -401,17 +394,23 @@ mod pallet {
             // to repatriate successfully, e.g. taker gets a little bit less
             // it should always ensure that the maker's request (maker_fill) is fully filled
             let taker_fill = Self::get_taker_fill(&order_data, maker_fill)?;
+            let order_account = Self::order_account(order_id);
 
-            // if base asset: fund the full amount, but charge base asset fees from taker later
-            T::AssetManager::repatriate_reserved_named(
-                &Self::reserve_id(),
-                maker_asset,
-                &maker,
-                &taker,
-                taker_fill,
-                BalanceStatus::Free,
-            )?;
+            if T::AssetManager::ensure_can_withdraw(maker_asset, &order_account, taker_fill).is_ok()
+            {
+                T::AssetManager::transfer(maker_asset, &order_account, &taker, taker_fill)?;
+            } else {
+                T::AssetManager::repatriate_reserved_named(
+                    &Self::reserve_id(),
+                    maker_asset,
+                    &maker,
+                    &taker,
+                    taker_fill,
+                    BalanceStatus::Free,
+                )?;
+            }
```

Comment:
Assume someone knows the order_account address and sends the maker_asset to it. The pallet would now try to fill the order using the transfer, although the maker has the maker_asset reserved. 

You could use the `reserved_balance` api function to check for the maker's balance instead.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1283/commits/e8cbf4111de693d759a51f52a50d27c50299c822